dspace-api/pom.xml

```diff
@@ -537,6 +618,35 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>org.apache.lucene</groupId>
+            <artifactId>lucene-core</artifactId>
+            <version>${solr.client.version}</version>
```

Comment:
Should this have a `<scope>test</scope>` as well? 

I'm not exactly sure why this new dependency was added, as it doesn't appear on `main` branch at all.  I'm assuming it's a dependency convergence error in the test environment?

Reply:
org.dspace.app.util.IndexVersion (in src/main) imports five classes from this artifact.  I probably fixed it up while using `mvn dependency:analyze` to track down an unrelated dependency puzzle.  It should be default-scoped.

Reply:
Ah, OK then. This seems fine then.  I was thrown off by the fact that lucene-core is in the midst of a bunch of other `test` dependencies.  At some point, we might want to put all the test dependencies together, separate from the non-test ones.  But, that can come in a followup PR.